[docs] Encourage DataGrid in /components/tables/ over alternatives (#22637)
mui-org/material-ui: docs/src/pages/components/tables/tables.md

When including tools, they should be placed directly above or below the table.

## Basic table

A simple example with no frills.

{{"demo": "pages/components/tables/BasicTable.js", "bg": true}}

## Dense table

A simple example of a dense table with no frills.

{{"demo": "pages/components/tables/DenseTable.js", "bg": true}}

## Sorting & selecting

This example demonstrates the use of `Checkbox` and clickable rows for selection, with a custom `Toolbar`. It uses the `TableSortLabel` component to help style column headings.

The Table has been given a fixed width to demonstrate horizontal scrolling. In order to prevent the pagination controls from scrolling, the TablePagination component is used outside of the Table. (The ['Custom Table Pagination Action' example](#custom-pagination-actions) below shows the pagination within the TableFooter.)

{{"demo": "pages/components/tables/EnhancedTable.js", "bg": true}}

## Data table

The `Table` components are meant to have a close mapping to the native `<table>` elements.
This constraint makes building rich data tables challenging.

> You can find the [`DataGrid` component](/components/data-grid/) for use cases that are focused around handling a large amount of tabular data.

Reviewer: ```suggestion
The [`DataGrid` component](/components/data-grid/) is designed for use-cases that are focused around handling a large amounts of tabular data.
```
